src/sage/combinat/set_partition.py

```diff
@@ -1873,8 +1873,7 @@ def arcs(self):
         arcs = []
         for p in self:
             p = sorted(p)
-            for i in range(len(p) - 1):
-                arcs.append((p[i], p[i + 1]))
+            arcs.extend((p[i], p[i + 1]) for i in range(len(p) - 1))
         return arcs
```

Comment:
Alternatively:
```
return [(p[i], p[i + 1]) for b in self if (p := sorted(b)) for i in range(len(p) - 1)]
```
but I'm not sure whether I like it better.

Reply:
I would rather keep it like I made it